crates/walrus-sdk/src/client/communication/node.rs

```diff
@@ -486,21 +510,72 @@ impl NodeWriteCommunication {
                 "the sliver is sufficiently small not to require a status check; \
                 storing the sliver",
             );
-        } else if self.get_sliver_status::<A>(blob_id, pair_index).await?
-            == StoredOnNodeStatus::Nonexistent
-        {
-            print_debug("the sliver is not stored on the node; storing the sliver");
         } else {
-            tracing::debug!(
-                ?pair_index,
-                sliver_type=?A::sliver_type(),
-                sliver_len=sliver.len(),
-                "the sliver is already stored on the node"
-            );
-            return Ok(());
+            match self.get_sliver_status::<A>(blob_id, pair_index).await {
+                Ok(StoredOnNodeStatus::Nonexistent) => {
+                    print_debug("the sliver is not stored on the node; storing the sliver");
+                }
+                Ok(_) => {
+                    tracing::debug!(
+                        ?pair_index,
+                        sliver_type=?A::sliver_type(),
+                        sliver_len=sliver.len(),
+                        "the sliver is already stored on the node"
+                    );
+                    return Ok(());
+                }
+                Err(error) => {
+                    return Err(error);
+                }
+            }
         }
 
-        self.store_sliver(blob_id, sliver, pair_index).await
+        let start = Instant::now();
+        match self.store_sliver(blob_id, sliver, pair_index).await {
+            Ok(()) => {
+                let completed_at = Instant::now();
+                if let Some(handle) = &self.auto_tune_handle {
+                    handle.record_success(sliver.len(), A::sliver_type(), completed_at);
+                }
```

Comment:
Question: Is there a reason to have this outside of `self.record_throughput_success`?

Reply:
no, not really. Fixed